functests/ptp/ptp.go

```diff
@@ -108,4 +90,397 @@ var _ = Describe("ptp", func() {
+		if runningPod.Spec.NodeName == nodeList.Items[NodeNumber].Name {
+			return true
+		}
+	}
+	return false
+}
+
+func compareOffsetTime(timeDiff string) bool {
+	var timeStampList []int
+	for _, line := range strings.Split(timeDiff, "\n") {
+		if strings.Contains(line, "openshift_ptp_max_offset_from_master") && !strings.Contains(line, "# ") {
+			lineValues := strings.Split(line, " ")
+			lastValue := strings.Trim(lineValues[len(lineValues)-1], "\r")
+			offsetFromMaster, err := strconv.Atoi(lastValue)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(offsetFromMaster).To(BeNumerically("<", 100))
+			Expect(offsetFromMaster).To(BeNumerically(">", -100))
+			timeStampList = append(timeStampList, offsetFromMaster)
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+// This function parses ethtool command output and detect interfaces which supports ptp protocol
+func isPTPEnabled(ethToolOutput *bytes.Buffer) bool {
+	var RxEnabled bool
+	var TxEnabled bool
+	var RawEnabled bool
+
+	scanner := bufio.NewScanner(ethToolOutput)
+	for scanner.Scan() {
+		line := strings.TrimPrefix(scanner.Text(), "\t")
+		parts := strings.Fields(line)
+		if parts[0] == _ETHTOOL_HARDWARE_RECEIVE_CAP {
+			RxEnabled = parts[1] == _ETHTOOL_RX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_TRANSMIT_CAP {
+			TxEnabled = parts[1] == _ETHTOOL_TX_HARDWARE_FLAG
+		}
+		if parts[0] == _ETHTOOL_HARDWARE_RAW_CLOCK_CAP {
+			RawEnabled = parts[1] == _ETHTOOL_RAW_HARDWARE_FLAG
+		}
+	}
+	return RxEnabled && TxEnabled && RawEnabled
+}
+
+func ptpDiscoveredInterfaceList(path string) []string {
+	var ptpInterfaces []string
+	var nodePtpDevice ptpv1.NodePtpDevice
+	fg, err := client.Client.CoreV1Interface.RESTClient().Get().AbsPath(path).DoRaw()
+	Expect(err).ToNot(HaveOccurred())
+
+	err = json.Unmarshal(fg, &nodePtpDevice)
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, intConf := range nodePtpDevice.Status.Devices {
+		ptpInterfaces = append(ptpInterfaces, intConf.Name)
+	}
+	return ptpInterfaces
+}
+
+func mutateProfile(profile ptpv1.PtpConfig, profileName string, nodeName string) *ptpv1.PtpConfig {
+	mutatedConfig := profile.DeepCopy()
+	priority := int64(0)
+	mutatedConfig.ObjectMeta.Reset()
+	mutatedConfig.ObjectMeta.Name = "test"
+	mutatedConfig.ObjectMeta.Namespace = ptpLinuxDaemonNamespace
+	mutatedConfig.Spec.Profile[0].Name = &profileName
+	mutatedConfig.Spec.Recommend[0].Priority = &priority
+	mutatedConfig.Spec.Recommend[0].Match[0].NodeLabel = nil
+	mutatedConfig.Spec.Recommend[0].Match[0].NodeName = &nodeName
+	mutatedConfig.Spec.Recommend[0].Profile = &profileName
+	return mutatedConfig
+}
+
+func waitUntilLogIsDetected(pod v1core.Pod, timeout time.Duration, neededLog string) {
+	Eventually(func() string {
+		logs, _ := pods.GetLog(&pod)
+		return logs
+	}, timeout*time.Second, 1*time.Second).Should(ContainSubstring(neededLog), fmt.Sprint("Timeout to detect log \"%s\" in pod \"%s\"", neededLog, pod.Name))
```

Comment:
This can now be just "timeout", passing 60 * time.Second outside (or even better, time.Minute)